src/ape/utils.py

```diff
@@ -394,9 +394,11 @@ def get_repo(self, repo_path: str) -> GithubRepository:
         """
 
         if repo_path not in self._repo_cache:
-            self._repo_cache[repo_path] = self._client.get_repo(repo_path)
-
-        return self._repo_cache[repo_path]
+            try:
+                self._repo_cache[repo_path] = self._client.get_repo(repo_path)
+                return self._repo_cache[repo_path]
```

Comment:
Ah, this is wrong because of the `if` statement.

To get around this, you could inver the if statement on line 396